oci/manager.go

```diff
@@ -123,6 +130,19 @@ func (manager *ociManager) Catalog(ctx context.Context, qopts ...packmanager.Que
 	query := packmanager.NewQuery(qopts...)
 	qname := query.Name()
 	qversion := query.Version()
+	platform := &v1.Platform{}
+
+	if query.Architecture() != "" {
+		platform.Architecture = query.Architecture()
+	}
+
+	if query.OS() != "" {
+		platform.OS = query.OS()
+	}
+
+	if query.OSVariant() != "" {
+		platform.Variant = query.OSVariant()
+	}
```

Comment:
No need to use `OSVariant` imho. Instead `OSVersion` and `OSFeatures`. 

We should use semver to do a comparison between `OSVersion` and the requested version.

We retrofit KConfig options into OSVariant. We should use `KConfig` instead and turn this into a slice representing mappings in the form`CONFIG_XYZ=y` for example

Reply:
sure thing, but I think it won't be possible to select based on it, or at least not before pulling